doc/specs/stdlib_math.md

```diff
@@ -91,6 +91,52 @@ program demo_clip_real
 end program demo_clip_real
 ```
 
+### `gcd` function
+
+#### Description
+
+Returns the greatest common divisor of two integers.
+
+#### Syntax
+
+`res = [[stdlib_math(module):gcd(interface)]] (a, b)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument(s)
+
+`a`: An integer.
+`b`: Another integer.
+
+Note: All arguments must be integers of the same `kind`.
+
+#### Output value or Result value
+
+The output is a scalar integer of the same `kind` as to that of the arguments.
+
+#### Examples
+
+##### Example 1:
+
+```fortran
+program demo_gcd
+  use stdlib_math, only: gcd
+  implicit none
+  integer :: a, b, c
+  
+  a = 48
+  b = 18
+  c = gcd(a, b)
+  ! c == 6
```

Comment:
I am trying to think like a beginner for whom it is the first programming language.

if `==` is accepted as a .true. statement then it is fine, I personally would have preferred `<--` (or any other random text) to indicate that the value of c is 6 because `==` operator, I think, is little ambiguous as it can return `.false.` also. Your call, shouldn't be a problem either way.

Reply:
We could also write it something like this:

```
c = gcd(a, b) ! returns 6
```

Reply:
I find arrows confusing. I like the return version.